x/cellarfees/keeper/hooks_test.go

```diff
@@ -0,0 +1,200 @@
+	}
+	cellarID := common.HexToAddress(event.EthereumSender)
+	feeDenom := "gravity0x1111111111111111111111111111111111111111"
+	cellarfeesKeeper.SetFeeAccrualCounters(ctx, cellarfeesTypes.FeeAccrualCounters{
+		Counters: []cellarfeesTypes.FeeAccrualCounter{
+			{
+				Denom: feeDenom,
+				Count: 1,
+			},
+		},
+	})
+	expectedCounters := cellarfeesTypes.FeeAccrualCounters{
+		Counters: []cellarfeesTypes.FeeAccrualCounter{
+			{
+				Denom: feeDenom,
+				Count: 0,
+			},
+		},
+	}
+
+	// mocks
+	suite.corkKeeper.EXPECT().HasCellarID(ctx, cellarID).Return(true)
+	suite.gravityKeeper.EXPECT().ERC20ToDenomLookup(ctx, common.HexToAddress(event.TokenContract)).Return(false, feeDenom)
+	suite.accountKeeper.EXPECT().GetModuleAccount(ctx, cellarfeesTypes.ModuleName).Return(feesAccount)
+	suite.auctionKeeper.EXPECT().GetActiveAuctions(ctx).Return([]*auctionTypes.Auction{})
+	suite.bankKeeper.EXPECT().GetBalance(ctx, feesAccount.GetAddress(), feeDenom).Return(sdk.NewCoin(feeDenom, event.Amount))
+	suite.auctionKeeper.EXPECT().BeginAuction(ctx, gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).Times(1)
+
+	require.NotPanics(func() { hooks.AfterSendToCosmosEvent(ctx, event) })
+	require.Equal(expectedCounters, cellarfeesKeeper.GetFeeAccrualCounters(ctx))
+}
+
+func (suite KeeperTestSuite) TestHooksCountAccruesAuctionDoesNotStart() {
+	ctx, cellarfeesKeeper, require := suite.ctx, suite.cellarfeesKeeper, suite.Require()
+	suite.SetupHooksTests(ctx, cellarfeesKeeper)
+
+	hooks := Hooks{k: cellarfeesKeeper}
+	event := gravitytypes.SendToCosmosEvent{
+		CosmosReceiver: feesAccount.GetAddress().String(),
+		Amount:         sdk.OneInt(),
+		EthereumSender: "0x0000000000000000000000000000000000000000",
+		TokenContract:  "0x1111111111111111111111111111111111111111",
+	}
+	cellarID := common.HexToAddress(event.EthereumSender)
+	feeDenom := "gravity0x1111111111111111111111111111111111111111"
+	cellarfeesKeeper.SetFeeAccrualCounters(ctx, cellarfeesTypes.FeeAccrualCounters{
+		Counters: []cellarfeesTypes.FeeAccrualCounter{
+			{
+				Denom: feeDenom,
+				Count: 1,
+			},
+		},
+	})
+	expectedCounters := cellarfeesTypes.FeeAccrualCounters{
+		Counters: []cellarfeesTypes.FeeAccrualCounter{
+			{
+				Denom: feeDenom,
+				Count: 2,
+			},
+		},
+	}
+
+	// mocks
+	suite.corkKeeper.EXPECT().HasCellarID(ctx, cellarID).Return(true)
+	suite.gravityKeeper.EXPECT().ERC20ToDenomLookup(ctx, common.HexToAddress(event.TokenContract)).Return(false, feeDenom)
+	suite.accountKeeper.EXPECT().GetModuleAccount(ctx, cellarfeesTypes.ModuleName).Return(feesAccount)
+	suite.auctionKeeper.EXPECT().GetActiveAuctions(ctx).Return([]*auctionTypes.Auction{})
+	suite.bankKeeper.EXPECT().GetBalance(ctx, feesAccount.GetAddress(), feeDenom).Return(sdk.NewCoin(feeDenom, event.Amount))
+	suite.auctionKeeper.EXPECT().BeginAuction(ctx, gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(auctionTypes.ErrDenomCannotBeEmpty).Times(1)
```

Comment:
I feel like I'm missing something reading this, what differentiates this test from the one preceding it that causes this error to be returned? This error type is when a `TokenPrice` fails `ValidateBasic`, which it seems like should not be happening? I also don't see either test setting token prices so I'm a bit confused.

Reply:
I needed an error that wouldn't cause a panic so I picked this one. The goal was just hit the branch that would increment a counter without resetting it because the auction failed to start but didn't panic. I can add better context with a comment?

Reply:
That might be helpful, although I am curious as to how that error is happening at all.

Reply:
You mean how it could happen or how it's happening in the test?